client/src/components/WorkflowInvocationState/WorkflowStepIcon.vue

```diff
@@ -0,0 +1,22 @@
+<script lang="ts" setup>
+import { computed } from "vue";
+
+import WorkflowIcons from "@/components/Workflow/icons";
+
+interface WorkflowStepIconProps {
+    stepType: "tool" | "data_input" | "data_collection_input" | "subworkflow" | "parameter_input" | "pause";
+}
+
+const props = defineProps<WorkflowStepIconProps>();
+
+const stepIcon = computed(() => {
+    return WorkflowIcons[props.stepType];
+});
+const cssClasses = computed(() => {
+    return ["fa", "mr-1", stepIcon.value];
+});
+</script>
+
+<template>
+    <i :class="cssClasses"></i>
+</template>
```

Comment:
```suggestion
const stepIcon = computed(() => {
    return WorkflowIcons[props.stepType];
});
</script>

<template>
    <FontAwesomeIcon class="mr-1" :icon="stepIcon" />
</template>
```

\+ adding library.add calls to `@/components/Workflow/icons`

Reply:
The library adding approach and the approach used by that class seem to have an incompatible set of icons available. I did my best to find the closest things and the class now includes two variants for the two ways the icons are used.

Reply:
I think the `<i>` approach has an older version of font-awesome. We would like to remove it in the long-run, as it bloats the bundle size. In case you can not find any compatible icon in the newer font-awesome version, you can place the svg of the old icon in an [icon pack](https://github.com/galaxyproject/galaxy/tree/dev/client/icons) to use it with the new method.